src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectNodeSection.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    [Flags]
+    public enum CustomSectionAttributes
```

Comment:
I'm not putting much though into it right now - but maybe we can expose this as higher level concepts and only convert to an enum in the object writer? That way we don't need two enums. But two enums are not necessarily terrible either, just obtuse.